### ISSUE TITLE
OS X installation: complex<double>::imag() and ::real()

### ISSUE THREAD

**Issue by defjaf** [healpy/healpy#284]
I am trying to `pip install healpy` with v 1.9.0 on OS X El Capitan, using a framework python build (hence, I think, I am constrained to use Apple clang/clang++ which is equivalent to gcc 4.2). Unfortunately, this means that I've changed both the healpy version and the OS X version at the same time. 

I now get the following errors from the sdist build 

        healpy/src/_healpy_sph_transform_lib.cc:923:27: error: no matching member function for call to 'imag'
                  almalms[i](l,0).imag(0.0);
                  ~~~~~~~~~~~~~~~~^~~~
        /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/c++/4.2.1/complex:1207:20: note: candidate function not viable: requires 0 arguments, but 1 was provided
      complex<double>::imag()
                       ^

and similar for ::real(), here and a few other places (4 errors all together).

**Comment by mreineck** [comment]
This is because the real() and imag() methods taking one argument are specific to C++11 ... but clang++ should support these without any problems. Which version of clang++ are you running? Or are you perhaps using flags like "-std=c++98"?

**Comment by mreineck** [comment]
Sorry ... I just see that under the hood your compilation indeed uses g++ 4.2.1.  That is most likely really too old. Is it possible to compile the package with "real" clang++ (which should not rely on g++ at all)?

**Comment by defjaf** [comment]
Hi Martin, 

Yes, that was the problem. I got confused because the new Xcode and OS X install meant that the command-line tools hadn't been updated completely, so compilers weren't able to reliably find header files. For info, what I did was

* `xcode-select --install` to make sure the compilation environment is ok
* `brew install llvm36` to get a more recent compiler (I assume other possibilities will work)
  * this compiler then requires 
```csh
     setenv CXX "clang++-3.6 -stdlib=libc++"
     setenv CXXFLAGS "-nostdinc++ -I/usr/local/opt/llvm36/lib/llvm-3.6/include/c++/v1"
     setenv CPPFLAGS "-I/usr/local/opt/llvm36/lib/llvm-3.6/include/c++/v1"
     setenv LDFLAGS -L/usr/local/opt/llvm36/lib/llvm-3.6/lib
```

**Comment by ngoldbaum** [comment]
I see a similar error in #292.  FWIW, I think making healpy compile using the version of clang that comes with OSX is worth doing. Asking people to install a new compiler version is a pretty steep requirement.

Also, it's not accurate that modern OSX comes with gcc-4.2. Apple stopped shipping GCC 4.2 with OSX 10.7. They now ship LLVM/Clang. On my laptop, I have:

```
$ clang --version
Apple LLVM version 7.0.0 (clang-700.0.72)
Target: x86_64-apple-darwin14.5.0
Thread model: posix
```

**Comment by lpsinger** [comment]
> I see a similar error in #292. FWIW, I think making healpy compile using the version of clang that comes with OSX is worth doing. Asking people to install a new compiler version is a pretty steep requirement.

Healpy should build without errors on most common Mac OS compiler toolchains, including the stock clang toolchain that comes with recent versions of Xcode. You shouldn't need to install a separate compiler from Homebrew (though it should *also* work with Homebrew's compilers). Did you recently migrate your machine from an older version of Mac OS? Sometimes that leaves bizarre combinations of old and new compilers around.

This issue is only superficially related to #292, which has to do with API changes in recent versions of healpix_cxx. The problem here, on the other hand, is that the build should test for C++11 support much earlier and fail with a comprehensible error message if the compiler does not support C++11.

@mreineck, would you please apply the patch at https://gist.github.com/lpsinger/5051706 to the HEALPix trunk?

**Comment by defjaf** [comment]
I have indeed just updated to El Capitan, right before this. But I have seen similar issues with other python source packages under older versions. But I am willing to believe I do have something problematic in my setup (although this actually seems less likely under the new rootless system which shouldn't have user-created files anywhere outside of /usr/local/.

So, just to confirm: the latest released healpy source *can* be compiled with the stock OS X clang toolchain, with framework builds of python from python.org?

Let me be even more specific: *has* this been done? (I.e., not "it should work in theory", but it actually does work!)

**Comment by mreineck** [comment]
Actually, I hadn't planned to require C++11 in Healpix 3.30 already, only in future versions, but the change in the interface of std::complex slipped by me :(
@lpsinger, I can certainly add the autoconf test, but if I have a day or two, I should also be able to replace the problematic C++11 feature by something that also works if the compiler only understands C++98. Would that be worthwhile?

**Comment by lpsinger** [comment]
> So, just to confirm: the latest released healpy source can be compiled with the stock OS X clang toolchain, with framework builds of python from python.org?

It works with the stock Python interpreter from Apple and the default Xcode compilers on Mac OS X 10.10.5. I just checked this; see the output in [this gist](https://gist.github.com/lpsinger/ef57849b6877db5506ed).

I do not have access to an El Capitan system at the moment, and I have never tried the python.org builds. The stock Apple interpreter is just a symlink into a framework:

```bash
$ ls -l /usr/bin/python2.7
lrwxr-xr-x  1 root  wheel  75 Oct  9  2014 /usr/bin/python2.7 -> ../../System/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7
```

So you are in slightly uncharted territory on El Capitan and a python.org build. You'll have to help us out here.

**Comment by lpsinger** [comment]
Oh, it also builds fine with the Python build from MacPorts, Anaconda, or Canopy. So if it doesn't work with python.org builds, that's very interesting and a strong clue.

**Comment by lpsinger** [comment]
> @lpsinger, I can certainly add the autoconf test, but if I have a day or two, I should also be able to replace the problematic C++11 feature by something that also works if the compiler only understands C++98. Would that be worthwhile?

I'm pretty confident we'll get to the bottom of @defjaf's issue, so no, I don't think it is necessary to add a C++98 fallback.

I'm reopening this issue to track progress on the C++11 configure check.

**Comment by defjaf** [comment]
OK, following @lpsinger, I used virtualenv for a clean environment, and tried both the system python and the python.org builds. 

System python works, [just as in the gist](https://gist.github.com/lpsinger/ef57849b6877db5506ed). But I get the same C++-11 clang errors with the python.org build as before.

(By the way, I had never used virtualenv before -- very cool!)

**Comment by lpsinger** [comment]
> System python works, just as in the gist. But I get the same C++-11 clang errors with the python.org build as before.

Would you please post the full transcript of that shell session in a gist?

**Comment by defjaf** [comment]
Here's [the attempt in a gist](https://gist.github.com/defjaf/1023df2b1315c34d574d).

**Comment by lpsinger** [comment]
Odd. There's nothing too strange in the compiler invocation, definitely nothing about C++ standards versions or SDKs:

      /usr/bin/clang -fno-strict-aliasing -fno-common -dynamic -arch i386 -arch x86_64 -g -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/jaffe/home/local/hp2/lib/python2.7/site-packages/numpy/core/include -I/private/var/folders/bk/zdhpxqj14y5fzcvlpybb4b640000gn/T/pip-build-Dz2fle/healpy/build/temp.macosx-10.6-intel-2.7/include -I/private/var/folders/bk/zdhpxqj14y5fzcvlpybb4b640000gn/T/pip-build-Dz2fle/healpy/build/temp.macosx-10.6-intel-2.7/include/healpix_cxx -I/private/var/folders/bk/zdhpxqj14y5fzcvlpybb4b640000gn/T/pip-build-Dz2fle/healpy/build/temp.macosx-10.6-intel-2.7/include -I/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c healpy/src/_healpy_sph_transform_lib.cc -o build/temp.macosx-10.6-intel-2.7/healpy/src/_healpy_sph_transform_lib.o

Do you get an error message if you compile the following program with `clang++`?

```c++
// test.cpp
// Compile with:
//     /usr/bin/clang++ test.cpp
#include <complex>
int main(int argc, char **argv)
{
    std::complex<double> a;
    a.imag(0.0);
    return 0;
}
```

**Comment by defjaf** [comment]
No errors, works fine!

```
snowleopard:~% /usr/bin/clang++ -v test.cpp -o test
Apple LLVM version 7.0.0 (clang-700.0.72)
Target: x86_64-apple-darwin15.0.0
Thread model: posix
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple x86_64-apple-macosx10.11.0 -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -main-file-name test.cpp -mrelocation-model pic -pic-level 2 -mthread-model posix -mdisable-fp-elim -masm-verbose -munwind-tables -target-cpu core2 -target-linker-version 253.2 -v -dwarf-column-info -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0 -stdlib=libc++ -fdeprecated-macro -fdebug-compilation-dir /Users/jaffe -ferror-limit 19 -fmessage-length 178 -stack-protector 1 -mstackrealign -fblocks -fobjc-runtime=macosx-10.11.0 -fencode-extended-block-signature -fcxx-exceptions -fexceptions -fmax-type-align=16 -fdiagnostics-show-option -fcolor-diagnostics -o /var/folders/bk/zdhpxqj14y5fzcvlpybb4b640000gn/T/test-b5365c.o -x c++ test.cpp
clang -cc1 version 7.0.0 based upon LLVM 3.7.0svn default target x86_64-apple-darwin15.0.0
ignoring nonexistent directory "/usr/include/c++/v1"
#include "..." search starts here:
#include <...> search starts here:
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /usr/include
 /System/Library/Frameworks (framework directory)
 /Library/Frameworks (framework directory)
End of search list.
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld" -demangle -dynamic -arch x86_64 -macosx_version_min 10.11.0 -o test /var/folders/bk/zdhpxqj14y5fzcvlpybb4b640000gn/T/test-b5365c.o -lc++ -lSystem /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.0.0/lib/darwin/libclang_rt.osx.a
```

**Comment by lpsinger** [comment]
> No errors, works fine!

What happens if you imitate the clang invocation that failed?

```sh
/usr/bin/clang -fno-strict-aliasing -fno-common -dynamic -arch i386 -arch x86_64 -g -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes test.cpp
```

**Comment by mreineck** [comment]
Don't forget to add the "-x c++" flag, otherwise this compilation will most likely fail!

**Comment by defjaf** [comment]
It works (even without `-x c++`; I think it uses the filename extension)!

Curiouser and curiouser.

**Comment by defjaf** [comment]
I think there is some information on why this is happening if you search for "python.org osx python gcc-4.2". My very, very, very vague understanding is that for backward compatibility they compile python with the equivalent of the gcc-4.2 compilers, which then enforces a default compile for extensions with that same compiler. But of course what I *don't* understand is how it is actually getting set up so that it points to those g++-4.2.1 includes. It might (?) be from an include that is being picked up from the python install itself?

What I don't understand is exactly *how* this is happening given the output we're seeing from the build.

Some refs:

* https://bugs.python.org/issue13590

**Comment by lpsinger** [comment]
Oh, crap. Does it work if you comment out these two lines?

https://github.com/healpy/healpy/blob/1.9.0/setup.py#L36-L37

**Comment by defjaf** [comment]
Hmmm, I haven't installed from source in a long time, but when I try, I get an  error

    ValueError: insecure string pickle

before it does anything interesting (this happens whether I try `python setup.py build`, or `pip install .`.

**Comment by lpsinger** [comment]
I've never seen that error message before. Would you please provide more context?

**Comment by defjaf** [comment]
Sorry:

```
snowleopard:~...dev/healpy% python setup.py build
running build
running build_py
creating build
creating build/lib.macosx-10.6-intel-2.7
creating build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/__init__.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/pixelfunc.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/sphtfunc.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/visufunc.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/fitsfunc.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/projector.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/rotator.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/projaxes.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/version.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/cookbook.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/newvisufunc.py -> build/lib.macosx-10.6-intel-2.7/healpy
copying healpy/zoomtool.py -> build/lib.macosx-10.6-intel-2.7/healpy
creating build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/__init__.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/test_doctest_cython.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/test_fitsfunc.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/test_pixelfunc.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/test_query_disc.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/test_sphtfunc.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/test_spinfunc.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
copying healpy/test/test_visufunc.py -> build/lib.macosx-10.6-intel-2.7/healpy/test
creating build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0002.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0004.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0008.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0016.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0032.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0064.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0128.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0256.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n0512.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n1024.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n2048.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n4096.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/pixel_window_n8192.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00002.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00004.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00008.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00016.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00032.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00064.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00128.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00256.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n00512.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n01024.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n02048.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n04096.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/weight_ring_n08192.fits -> build/lib.macosx-10.6-intel-2.7/healpy/data
copying healpy/data/totcls.dat -> build/lib.macosx-10.6-intel-2.7/healpy/data
creating build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/cl_wmap_band_iqumap_r9_7yr_W_v4_udgraded32_II_lmax64_rmmono_3iter.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/cl_wmap_band_iqumap_r9_7yr_W_v4_udgraded32_II_lmax64_rmmono_3iter_nomask.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/cl_wmap_band_iqumap_r9_7yr_W_v4_udgraded32_IQU_lmax64_rmmono_3iter.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/cl_wmap_band_iqumap_r9_7yr_WVxspec_v4_udgraded32_II_lmax64_rmmono_3iter.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/gaussbeam_10arcmin_lmax512_pol.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/map_synfast_seed12345.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/wmap_band_iqumap_r9_7yr_V_v4_udgraded32.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/wmap_band_iqumap_r9_7yr_V_v4_udgraded32_masked.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/wmap_band_iqumap_r9_7yr_W_v4_udgraded32.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/wmap_band_iqumap_r9_7yr_W_v4_udgraded32_masked.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/wmap_band_iqumap_r9_7yr_W_v4_udgraded32_masked_smoothed10deg_fortran.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/wmap_band_iqumap_r9_7yr_W_v4_udgraded32_smoothed10deg_fortran.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
copying healpy/test/data/wmap_temperature_analysis_mask_r9_7yr_v4_udgraded32.fits -> build/lib.macosx-10.6-intel-2.7/healpy/test/data
running build_clib
checking if configure script for library 'cfitsio' exists
checking if configure script for library 'healpix_cxx' exists
running 'autoreconf -i' for library 'healpix_cxx'
Traceback (most recent call last):
  File "setup.py", line 465, in <module>
    license='GPLv2'
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/command/build.py", line 127, in run
    self.run_command(cmd_name)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 971, in run_command
    cmd_obj.ensure_finalized()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 109, in ensure_finalized
    self.finalize_options()
  File "setup.py", line 196, in finalize_options
    check_call(['autoreconf', '-i'], cwd=build_info['local_source'])
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 535, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1334, in _execute_child
    child_exception = pickle.loads(data)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/pickle.py", line 1382, in loads
    return Unpickler(file).load()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/pickle.py", line 858, in load
    dispatch[key](self)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/pickle.py", line 966, in load_string
    raise ValueError, "insecure string pickle"
ValueError: insecure string pickle
```

**Comment by lpsinger** [comment]
That's weird. No idea what's doing that. One more question: to check my diagnosis, would you please fire up the python.org interpreter and post the output of the following command?

    >>> from distutils.sysconfig import get_config_var; print(get_config_var('MACOSX_DEPLOYMENT_TARGET'))

**Comment by defjaf** [comment]
Hi,

It gives 10.6 from the python.org build, and 10.11 from the built-in. I think that's the point, since that's how the python.org builds are created, for backward compatibility.

Andrew

**Comment by lpsinger** [comment]
Hmm. The C++11 standard library, selected by the compiler options `-stdlib=libc++`, requires `MACOSX_DEPLOYMENT_TARGET=10.7` or later.

We could force Healpy to build against a different Mac OS X deployment target than the Python interpreter, but that would probably lead to all sorts of runtime linking errors (see #151).

**Comment by lpsinger** [comment]
@mreineck: what options do you have for making this not require C++11? This is likely going to be an issue for Python users on Mac OS for a while because there are several popular Python distributions that are still building against libstdc++.

**Comment by defjaf** [comment]
Aha, well this appears to "solve" the problem -- if I set MACOSX_DEPLOYMENT_TARGET=10.11 then it seems to pick the correct (newer) compiler and/or library and compiles fine. 

A cursory check seems to indicate things are working. Is there a proper test suite for me to make sure?

This then raises the question about whether this is the best compiler for the job, but at least it works...

**Comment by mreineck** [comment]
I opened pull request #297 to remove the C++11 features.

**Comment by defjaf** [comment]
Great, although this seems like a stopgap measure to some extent. Does the explicit MACOSX_DEPLOYMENT_TARGET=10.11 have any downside? It seems that some other issues imply that it might, at least with some distributions. 

Conversely, does it have an upside? Does it use a more modern compiler or allow greater optimisation?

**Comment by mreineck** [comment]
The libstdc++ present on OSX is quite ancient ... and it won't be updated because of licensing issues. Newer C++ standard libraries with support for C++11 can be made much more efficient, so in the future I certainly recommend switching.

**Comment by mreineck** [comment]
I updated the Healpix C++ svn repository and removed the C++11 features there as well. It is revision 808 on branches/branch_v330r807/.

**Comment by lpsinger** [comment]
@defjaf, did this help? We might want to do another minor release to pick up this and the Numpy 1.10.0 compatibility fix.

**Comment by defjaf** [comment]
Hi,

I haven’t tried the latest version — for some reason I have trouble building the master from the repo — but I am satisfied so far with explicitly setting MACOSX_DEPLOYMENT_TARGET, although I haven’t been able to get 100% confirmation that it’s safe!

I will at least try the next release with the default compilers/libraries to check.

Andrew

**Comment by zonca** [comment]
@defjaf can you please check if this is fixed with 1.9.1?

**Comment by defjaf** [comment]
Yes, it seems to work with python.org installs of 2.7.10 and 3.5

Yours,

Andrew
